spec.html

```diff
@@ -42482,104 +42631,222 @@ <h1>
         </dl>
         <emu-alg>
           1. Assert: The surrounding agent is in the critical section for _WL_.
-          1. Assert: _W_ is AgentSignifier().
-          1. Assert: _W_ is on the list of waiters in _WL_.
+          1. Assert: _WL_.[[Waiters]] contains _waiterRecord_.
+          1. Let _thisAgent_ be AgentSignifier().
+          1. Assert: _waiterRecord_.[[AgentSignifier]] is _thisAgent_.
+          1. Assert: _waiterRecord_.[[PromiseCapability]] is ~blocking~.
           1. Assert: AgentCanSuspend() is *true*.
-          1. Let _additionalTimeout_ be an implementation-defined non-negative mathematical value.
-          1. Let _timeout_ be _minimumTimeout_ + _additionalTimeout_.
-          1. NOTE: When _minimumTimeout_ is +∞, _timeout_ is also +∞ and the following step can terminate only by another agent calling NotifyWaiter.
-          1. Perform LeaveCriticalSection(_WL_) and suspend _W_ for up to _timeout_ milliseconds, performing the combined operation in such a way that a notification that arrives after the critical section is exited but before the suspension takes effect is not lost. _W_ can wake from suspension either because the timeout expired or because it was notified explicitly by another agent calling NotifyWaiter with arguments _WL_ and _W_, and not for any other reasons at all.
+          1. Perform LeaveCriticalSection(_WL_) and suspend the surrounding agent until the time is _waiterRecord_.[[TimeoutTime]], performing the combined operation in such a way that a notification that arrives after the critical section is exited but before the suspension takes effect is not lost. The only way for the surrounding agent to wake from suspension is by another agent calling NotifyWaiter with arguments _WL_ and _thisAgent_. This can happen if the timeout expires or via a call to `Atomics.notify`.
           1. Perform EnterCriticalSection(_WL_).
-          1. If _W_ was notified explicitly by another agent calling NotifyWaiter with arguments _WL_ and _W_, return *true*.
-          1. Return *false*.
+          1. Return ~unused~.
         </emu-alg>
-        <emu-note>
-          <p>_additionalTimeout_ allows implementations to pad timeouts as necessary, such as for reducing power consumption or coarsening timer resolution to mitigate timing attacks. This value may differ from call to call of SuspendAgent.</p>
-        </emu-note>
       </emu-clause>
 
       <emu-clause id="sec-notifywaiter" type="abstract operation">
         <h1>
           NotifyWaiter (
-            _WL_: a WaiterList,
-            _W_: an agent signifier,
+            _WL_: a WaiterList Record,
+            _waiterRecord_: a Waiter Record,
           ): ~unused~
         </h1>
         <dl class="header">
         </dl>
         <emu-alg>
           1. Assert: The surrounding agent is in the critical section for _WL_.
-          1. Notify the agent _W_.
+          1. Assert: _waiterRecord_.[[Result]] is either *"ok"* or *"timed-out"*.
```

Comment:
This doesn't narrow the type at all, since those are the only values allowed for `[[Result]]`. I don't think this assert is helpful here.

Reply:
Removed.